chsdi/tests/functional/test_helpers.py

```diff
@@ -4,13 +4,33 @@
 
 from pyramid import testing
 
-from chsdi.lib.helpers import make_agnostic, make_api_url, check_url, transformCoordinate, sanitize_url, check_even, round, format_search_text, remove_accents, escape_sphinx_syntax, quoting, float_raise_nan, resource_exists, parseHydroXML
+from chsdi.lib.helpers import make_agnostic, make_api_url, check_url, transformCoordinate, sanitize_url, check_even, round, format_search_text, remove_accents, escape_sphinx_syntax, quoting, float_raise_nan, resource_exists, parseHydroXML, locale_negotiator, versioned
 
 from urlparse import urljoin
 
 
 class Test_Helpers(unittest.TestCase):
 
+    def test_versioned(self):
+        import os.path
+        request = testing.DummyRequest()
+        request.host = 'api3.geo.admin.ch'
+        request.scheme = 'http'
+        request.registry.settings = {}
+        request.registry.settings['apache_base_path'] = 'main'
+
+        request.registry.settings['app_version'] = None
+        request.registry.settings['entry_path'] = 'https://api3.geo.admin.ch'
+        test_path = request.registry.settings['entry_path']
```

Comment:
hardcode test_path here. I see 2 use cases:
1. simple root url (http://something.com)
2.url with path (https://something.com/entry_path/more/to)
3) the above without http: